Source/DafnyCore/Resolver/NameResolutionAndTypeInference/NameResolutionAndTypeInference.cs

```diff
@@ -316,7 +316,49 @@ public partial class ModuleResolver {
         if (attr.Args != null) {
           foreach (var arg in attr.Args) {
             Contract.Assert(arg != null);
-            ResolveExpression(arg, resolutionContext);
+            if (!(Attributes.Contains(attributeHost.Attributes, "opaque_reveal") && attr.Name == "fuel" && arg is NameSegment)) {
+              ResolveExpression(arg, resolutionContext);
+            } else {
+              // Manually resolving NameSegments that are present in fuel attributes of reveal lemmas.
+              // This is because reveal lemmas are static and we want to allow a refence to non-static original procedures
+              // in static context in this setting.
+              //
+              // Most of the following code is copied from AnnotateRevealFunction() in OpaqueMemberRewriter.cs.
```

Comment:
Can you please put this branch in a separate function, so that you can put it as the function's comment?